### DISCUSSION TITLE
OpenZFS 2.1.4 on FreeBSD and xattr=sa

### DISCUSSION THREAD

**Discussion by CyberCr33p** [openzfs/zfs#14443] [Q&A] [answered]
I create a new dataset in a FreeBSD server using OpenZFS 2.1.4.

After creation I set `acltype=nfsv4` and `xattr=sa`

I create a new file and `setfacl -m u:www:x:allow file`

I `zdb` the file inode number but it doesn't show anything about xattr.

Shouldn't zdb show something about xattr?

**Comment by ghost** [comment, marked as answer]
Answered elsewhere as well but here again for posterity: unlike the posix ACLs for Linux, NFSv4 ACLs are not stored in xattrs.

**Reply by amotin** [reply]
Depending on settings xattrs can be stores either as separate files in a hidden directory, or as part of system attributes in dnode bonus buffer.  If bonus buffer does not have enough space, extra is leaking into separate spill block, referenced by a block pointer at the end of bonus buffer.
